llvm/docs/AMDGPUUsage.rst

```diff
@@ -4255,10 +4255,9 @@ same *vendor-name*.
                                                                   wavefront for
                                                                   GFX6-GFX9. A register
                                                                   is required if it is
-                                                                  used explicitly, or
+                                                                  written to, or
```

Comment:
nit: reading from a register should count as well (e.g. a function that reads arguments from high VGPRs, or a shader that reads hw-initialized registers and only writes to lower VGPRs).

Reply:
I think @jayfoad's point was that we don't need to count those because they're counted wherever they're defined. I tried to explain that in the new PR description, but let me know if you disagree with that reasoning (personally I'm also leaning towards being more precise, i.e. including the reads as well, but then I don't know how to reduce the usage for the init-whole-wave case without giving it special treatment).